addons/stock/models/stock_location.py

```diff
@@ -241,7 +241,14 @@ def _get_putaway_strategy(self, product, quantity=0, package=None, packaging=Non
             for values in quant_data:
                 qty_by_location[values['location_id'][0]] += values['quantity']
 
-        return putaway_rules._get_putaway_location(product, quantity, package, qty_by_location) or self
+        putaway_location = putaway_rules._get_putaway_location(product, quantity, package, qty_by_location) or self
+        # A putaway location might not be found, but a move_line shouldn't target a 'view' location. So search the first valid child.
+        if putaway_location.usage == 'view':
```

Comment:
The `_get_putaway_location` shouldn't return a view location (because of you new domain, and take only in account internal usage), Right ? If this condition `putaway_location.usage == 'view'` is True, then `putaway_location == self`, then you have already the result of you search in `locations` variable.

Reply:
To improve the readability (because it is hard to understand that `if putaway_location.usage == 'view'` then `putaway_location == self`, then ...: 
```python
putaway_location = putaway_rules._get_putaway_location(product, quantity, package, qty_by_location)
if not putaway_location:
   putaway_location = locations[0] if locations and self.usage == 'view' else self
```